axelor-base/src/main/java/com/axelor/apps/base/service/discount/GlobalDiscountServiceImpl.java

```diff
@@ -0,0 +1,100 @@
+package com.axelor.apps.base.service.discount;
+
+import com.axelor.apps.base.db.Currency;
+import com.axelor.apps.base.db.repo.PriceListLineRepository;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.HashMap;
+import java.util.Map;
+
+public class GlobalDiscountServiceImpl implements GlobalDiscountService {
+
+  @Override
+  public BigDecimal computeDiscountFixedEquivalence(
+      BigDecimal exTaxTotal, BigDecimal priceBeforeGlobalDiscount) {
+    if (exTaxTotal == null || priceBeforeGlobalDiscount == null) {
+      return BigDecimal.ZERO;
+    }
+    return priceBeforeGlobalDiscount.subtract(exTaxTotal);
+  }
+
+  @Override
+  public BigDecimal computeDiscountPercentageEquivalence(
+      BigDecimal exTaxTotal, BigDecimal priceBeforeGlobalDiscount) {
+    if (exTaxTotal == null || priceBeforeGlobalDiscount == null) {
+      return BigDecimal.ZERO;
+    }
+    return priceBeforeGlobalDiscount
+        .subtract(exTaxTotal)
+        .multiply(BigDecimal.valueOf(100))
+        .divide(priceBeforeGlobalDiscount, RoundingMode.HALF_UP);
+  }
+
+  @Override
+  public Map<String, Map<String, Object>> setDiscountDummies(
+      Integer discountTypeSelect,
+      Currency currency,
+      BigDecimal exTaxTotal,
+      BigDecimal priceBeforeGlobalDiscount) {
+    Map<String, Map<String, Object>> attrsMap = new HashMap<>();
+    switch (discountTypeSelect) {
+      case PriceListLineRepository.AMOUNT_TYPE_PERCENT:
+        setSaleOrderPercentageGlobalDiscountDummies(
+            currency, exTaxTotal, priceBeforeGlobalDiscount, attrsMap);
+        break;
+      case PriceListLineRepository.AMOUNT_TYPE_FIXED:
+        setSaleOrderFixedGlobalDiscountDummies(
+            currency, exTaxTotal, priceBeforeGlobalDiscount, attrsMap);
+        break;
+    }
+    return attrsMap;
+  }
+
+  protected void setSaleOrderPercentageGlobalDiscountDummies(
```

Comment:
I think you forgot to rename this method, because it's not related only to sale order now